lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -33,8 +34,47 @@ def overage(issue)
           end
 
           def process_file(file)
-            RubyParser.new.process(File.binread(file), file, TIMEOUT)
+            if File.extname(file) == ".erb"
```

Comment:
WDYT think about extracting all of this into a `Parser` class like the other languages have? Keeping that logic here made sense when it was one-liner, but I think this is complex enough that extracting makes sense.